eden/converter/graph/gspan.py

```diff
@@ -28,37 +24,67 @@ def gspan_to_eden(input=None, options=dict()):
         yield gspan_to_networkx(header, string_list)
 
 
-def gspan_to_networkx(header, string_list):
-    graph = nx.Graph()
-    graph.graph['id'] = header
-    for line in string_list:
-        if line.strip():
-            line_list = line.split()
-            fc = line_list[0]
-            # process vertices
-            if fc in ['v', 'V']:
-                vid = int(line_list[1])
-                vlabel = line_list[2]
-                # lowercase v indicates active viewpoint
-                if fc == 'v':
-                    weight = 1
-                else:  # uppercase v indicates no-viewpoint, in the new EDeN this is simulated via a smaller weight
-                    weight = 0.1
-                graph.add_node(vid, ID=vid, label=vlabel, weight=weight)
-                # extract the rest of the line  as a JSON string that contains all attributes
-                attribute_str = ' '.join(line_list[3:])
-                if attribute_str.strip():
-                    attribute_dict = json.loads(attribute_str)
-                    graph.node[vid].update(attribute_dict)
-            # process edges
-            if fc == 'e':
-                srcid = int(line_list[1])
-                destid = int(line_list[2])
-                elabel = line_list[3]
-                graph.add_edge(srcid, destid, label=elabel)
-                attribute_str = ' '.join(line_list[4:])
-                if attribute_str.strip():
-                    attribute_dict = json.loads(attribute_str)
-                    graph.edge[srcid][destid].update(attribute_dict)
+def gspan_to_networkx(header, lines):
+    """Take a string list in the extended gSpan format and returns a NetworkX graph.
+
```

Comment:
We need to stick to a proper syntax here:

    """Calculate the square root of a number.

    Args:
        n: the number to get the square root of.
    Returns:
        the square root of n.
    Raises:
        TypeError: if n is not a number.
        ValueError: if n is negative.

    """

https://pythonhosted.org/an_example_pypi_project/sphinx.html#full-code-example